src/filter-input.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * input Filter Widget
+ *
+ * The input filter data widget is a simple widget designed to display an input field allowing to filter
+ * the data that matches the text typed.
+ * As opposed to the other graphs, this doesn't display any result and doesn't update its display,
+ * it's just to input an filter other graphs.
+ *
+ * @class inputFilter
+ * @memberof dc
+ * @mixes dc.baseMixin
+ * @example
+ *
+ * var data = [{"firstname":"John","lastname":"Coltrane"}{"firstname":"Miles",lastname:"Davis"}]
+ * var ndx = crossfilter(data);
+ * var dimension = ndx.dimension(function(d) {
+ *     return d.lastname.toLowerCase() + ' ' + d.firstname.toLowerCase();
+ * });
+ *
+ * dc.inputFilter('#search')
+ *     .dimension(dimension);
+ *     // you don't need the group() function
+ *
+ * @param {String|node|d3.selection|dc.compositeChart} parent - Any valid
+ * {@link https://github.com/d3/d3-selection/blob/master/README.md#select d3 single selector}
+ * specifying a dom block element such as a div; or a dom element or d3 selection.
+ * @param {String} [chartGroup] - The name of the chart group this chart instance should be placed in.
+ * Interaction with a chart will only trigger events and redraws within the chart's group.
+ * @returns {dc.inputFilter}
+ **/
+
+dc.inputFilter = function (parent, chartGroup) {
+    var INPUT_CSS_CLASS = 'dc-filter-input';
+
+    var _chart = dc.baseMixin({});
+
+    var _normalize = function (s) {
+        return s.toLowerCase();
+    };
+
+    var _filterFunctionFactory = function (query) {
```

Comment:
The coding convention in dc.js is to use a function declaration rather than a variable, for local functions that won't be replaced. (Using a variable for `_normalize` makes sense because it will change.)

Reply:
Did you mean for `filterFunctionFactory`?

`filterFunctionFactory` can change in some specific cases. There is get/set for this.

Reply:
Whoops, you're so right. Sorry for the noise.